DUI3-DX/Connectors/ArcGIS/Speckle.Connectors.ArcGIS3/Operations/Receive/HostObjectBuilder.cs

```diff
@@ -105,21 +89,15 @@ CancellationToken cancellationToken
       {
         if (IsGISType(obj))
         {
-          var result = ConvertNativeLayers((Collection)obj, path);
-          convertedGISObjects.Add(result);
-          // NOTE: Dim doesn't really know what is what - is the result.path the id of the obj?
-          // TODO: is the type in here basically a GIS Layer?
-          results.Add(new(Status.SUCCESS, obj, result.path, "GIS Layer"));
+          string nestedLayerPath = $"{string.Join("\\", path)}\\{((Collection)obj).name}";
+          string datasetId = (string)_converter.Convert(obj);
```

Comment:
By any chance it returns non-string object? Is it likely?

Reply:
All native GIS layers during conversion will be written to a dataset and return its ID (more detailed explanation here: https://www.notion.so/speckle/ArcGIS-explainers-e3c2571d0208409fa98fe87d41247c2d)

Reply:
Good to know! Ty